### PR TITLE
feat: add serializer option

### LINKED ISSUE #3: How to parse value before send to URL?
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[ ] Feature request
[x] Documentation issue or request
[ ] Support request
[ ] Other... Please describe:
</code></pre>

## Current behavior
I have a `<input matDatepicker>` and when I change the date it sends the `Date` to the url.

## Expected behavior
I'd like to know if there's an option to _parse_ it (in this case to `string`) before send it to the url. I've already tried to use https://github.com/ngneat/bind-query-params#parser, however it seems that `parser` is only called on initialization.

## Minimal reproduction of the problem with instructions
https://stackblitz.com/edit/angular-ewwssg

## What is the motivation / use case for changing the behavior?
I'd like to have an ISO string in URL, but keeping Date in my `FormControl`.

## Environment

<pre><code>
Angular version: 11x
</pre></code>

Others:
Thanks for this awesome lib :pray:.

### PR DESCRIPTION
**Pull request by rafaelss95**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Currently there's no way to intercept value before persist them in URL.

Issue Number: Closes #3.

## What is the new behavior?
With the `serializer` option, it's possible now.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by rafaelss95** [comment]
@NetanelBasal while doing this, I just noticed that the `{ strict: true }` is not enabled in the repo and thus can cause some issues. Let me know if I can send a follow-up PR addressing this.

File: projects/ngneat/bind-query-params/src/lib/utils.ts [hunk 1]
```diff
@@ -27,21 +27,18 @@ export function get(obj: object, path: string): any {
...
-
-  const result = {};
-
-  toArray.forEach(({ value, queryKey }) => {
-    const isEmpty = value === null || value === undefined || !value.toString();
-    result[queryKey] = isEmpty ? null : value!.toString();
-  });
-
-  return result;
+  return coerceArray(params).reduce(
```
Comment (NetanelBasal) [inline projects/ngneat/bind-query-params/src/lib/utils.ts:35]: The prev code was intentional. I find it way more readable than this version.

**Comment by NetanelBasal** [comment]
You can change to strict true in another PR.

File: projects/ngneat/bind-query-params/src/lib/utils.ts (hunk 1 above)
Comment (rafaelss95) [reply projects/ngneat/bind-query-params/src/lib/utils.ts]: Let me know if it's okay now.

**Pull request merged**